tensorflow_probability/python/experimental/mcmc/particle_filter_test.py

```diff
@@ -238,23 +236,14 @@ def observation_fn(_, state):
                          batch_shape[0],
                          num_particles,
                          batch_shape[1]])
+    self.assertAllEqual(log_weights.shape,
+                        [num_timesteps,
+                         batch_shape[0],
+                         num_particles,
+                         batch_shape[1]])
     self.assertAllEqual(incremental_log_marginal_likelihoods.shape,
                         [num_timesteps] + batch_shape)
 
-    self.assertAllClose(
-        self.evaluate(
-            tf.reduce_sum(tf.exp(log_weights) *
-                          particles['position'], axis=2)),
-        observed_positions,
```

Comment:
I would hope this assertion would pass if we changed this line to `observed_positions[..., tf.newaxis, tf.newaxis]`.

But if that didn't work, we could also do: `tf.broadcast_to(observation_positions, [num_timesteps] + batch_shape)`.